src/Microsoft.Management.UI.Internal/ManagementList/FilterCore/FilterRules/SelectorFilterRule.cs

```diff
@@ -48,6 +48,26 @@ public SelectorFilterRule()
             this.AvailableRules.SelectedValueChanged += this.AvailableRules_SelectedValueChanged;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the SelectorFilterRule class from an existing instance.
+        /// </summary>
+        /// <param name="source">The source to initialize from.</param>
+        public SelectorFilterRule(SelectorFilterRule source) : base(source)
+        {
+            this.AvailableRules = new ValidatingSelectorValue<FilterRule>();
+            foreach (DataErrorInfoValidationRule rule in source.AvailableRules.ValidationRules)
+            {
+                this.AvailableRules.AddValidationRule(rule);
```

Comment:
Should we clone the rule too?
```suggestion
                this.AvailableRules.AddValidationRule(rule.Clone());
```

Reply:
DataErrorInfoValidationRule is stateless and not derived from FilterRule. Perhaps I should use different names for the loop variables to avoid confusion.

Reply:
In any case we should create new object to follow "DeepCopy".

Reply:
Thanks! I see. In any case we should create new object to follow "DeepCopy" as BinaryFormatter does.